python/ray/train/batch_predictor.py

```diff
@@ -287,7 +287,9 @@ def __call__(self, input_batch: DataBatchType) -> DataBatchType:
             if preprocessor:
                 # Set the in-predictor preprocessing to a no-op when using a separate
                 # GPU stage. Otherwise, the preprocessing will be applied twice.
-                override_prep = BatchMapper(lambda x: x)
+                override_prep = BatchMapper(
+                    lambda x: x, batch_format=predict_stage_batch_format
+                )
```

Comment:
Is it possible to add a unit test for this codepath? Test preprocessing before sending to a separate GPU stage to do prediction.

Reply:
this is being tested here https://sourcegraph.com/github.com/ray-project/ray/-/blob/python/ray/train/tests/test_batch_predictor.py?L104:5

Reply:
I see, I think the test was still pending and not showing up on the flaky tests earlier.